packages/hint-manifest-icons/src/hint.ts

```diff
@@ -0,0 +1,220 @@
+/**
+ * @fileoverview Checks usage of Web App Manifest icons
+ */
+import { URL } from 'url';
+import * as getImageData from 'image-size';
+import imageType, { ImageTypeResult } from 'image-type';
+import { IHint, NetworkData, HintContext } from 'hint';
+import { ManifestEvents, ManifestParsed, ManifestImageResource } from '@hint/parser-manifest';
+import { debug as d } from '@hint/utils';
+
+import meta from './meta';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class ManifestIconHint implements IHint {
+    public static readonly meta = meta;
+    public constructor(context: HintContext<ManifestEvents>) {
+        const getIconFullPath = (iconSrc: string, hostnameWithProtocol: string) => {
+            return `${hostnameWithProtocol}/${iconSrc}`;
+        };
+
+        /**
+         * See if the `icon` file actually
+         * exists and is accessible.
+         */
+        const iconExists = async (iconPath: string) => {
+            let networkData: NetworkData;
+
+            try {
+                networkData = await context.fetchContent(iconPath);
+            } catch (e) {
+                debug(`Failed to fetch the ${iconPath} file`);
+                const message = `Icon could not be fetched (request failed).`;
+
+                context.report(iconPath, message);
+
+                return null;
+            }
+            const response = networkData.response;
+
+            if (response.statusCode !== 200) {
+                const message = `Icon could not be fetched (status code: ${response.statusCode}).`;
+
+                context.report(iconPath, message);
+
+                return null;
+            }
+
+            return response.body.rawContent;
+        };
+        /**
+         * Passes only for the PNG files
+         * @param iconType type specified in the manifest file
+         * @param rawContent raw datastream
+         * @param iconPath icon resource path
+         */
+        const validateImageType = (iconType: string | undefined, rawContent: Buffer, iconPath: string) => {
+            const allowedTypes = ['png', 'jpg'];
+
+            if (iconType === undefined) {
+                const message = `Icon type was not specifed`;
+
+                context.report(iconPath, message);
+
+                return false;
+            }
+
+            const specifiedType = iconType.split('/')[1];
+
+            /** Handling for the corrupt rawContent */
+            if (rawContent !== null) {
+                const image: ImageTypeResult | null = imageType(rawContent);
```

Comment:
I usually try to avoid specifying types that get inferred correctly (keeps the code a bit closer to JS). Here we'll get the correct type from the function.
```suggestion
                const image = imageType(rawContent);
```

Reply:
Resolved. Learning TypeScript as I move forward with this codebase.